src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (C) 20211 Licensed under the Apache License, Version
+ * 2.0 (the "License"). You may not use this file except in
+ * compliance with the License. A copy of the License is located
+ * at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "OO_HostMem.h"
+#include "tools/common/XBUtilities.h"
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+#include <boost/algorithm/string.hpp>
+namespace po = boost::program_options;
+
+namespace {
+bool
+string2action(std::string str)
```

Comment:
This algorithm while it does work seems to be an overkill, why didn't you use boost::iequals().

Reply:
fixed